src/executable_code_finder/lib.rs

```
@@ -0,0 +1,241 @@
+        }
+
+        let interner = Arc::new(Mutex::new(interner));
+        let mut handles = vec![];
+
+        for (_, path_group) in path_groups {
+            let interner = interner.clone();
+            let bar = bar.clone();
+            let files_processed = files_processed.clone();
+            let logger = logger.clone();
+            let invalid_files = invalid_files.clone();
+            let executable_lines = executable_lines.clone();
+            let root_dir = config.root_dir.clone();
+
+            let handle = std::thread::spawn(move || {
+                let new_interner = ThreadedInterner::new(interner);
+
+                for file_path in &path_group {
+                    match scan_file(&new_interner, &root_dir, *file_path, &logger.clone(), ) {
+                        Err(_) => {
+                            invalid_files.lock().unwrap().push(*file_path);
+                        }
+                        Ok(res) => {
+                            let mut executable_lines = executable_lines.lock().unwrap();
+                            if !res.executable_lines.is_empty() {
+                                executable_lines.push(res);
+                            }
+                        }
+                    };
+
+                    let mut tally = files_processed.lock().unwrap();
+                    *tally += 1;
+
+                    update_progressbar(*tally, bar.clone());
+                }
+            });
+
+            handles.push(handle);
+        }
+
+        for handle in handles {
+            handle.join().unwrap();
+        }
+
+        if let Some(bar) = &bar {
+            bar.finish_and_clear();
+        }
+
+        if logger.can_log_timing() {
+            logger.log_sync(&format!(
+                "Scanning files took {:.2?}",
+                file_scanning_now.elapsed()
+            ));
+        }
+    }
+
+    let _invalid_files = Arc::try_unwrap(invalid_files)
```

You can get rid of this assignment

---

Maybe `panic!()` earlier when you see any invalid file.